modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java

```diff
@@ -1393,6 +1406,273 @@ private String _testAddSystemObjectField(
+			new PicklistObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).listTypeDefinitionId(
+				_listTypeDefinition.getListTypeDefinitionId()
+			).name(
+				"a" + RandomTestUtil.randomString()
+			).objectDefinitionId(
+				objectDefinition.getObjectDefinitionId()
+			).objectFieldSettings(
+				Arrays.asList(
+					_createObjectFieldSetting(
+						ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+						_listTypeEntryKey),
+					_createObjectFieldSetting(
+						ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+						ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE))
+			).build());
+
+		_assertObjectFieldSettingDefaultValue(
+			_listTypeEntryKey, ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE,
+			objectField);
+
+		_assertDefaultValue(_listTypeEntryKey, objectField);
+
+		_objectFieldLocalService.deleteObjectField(objectField);
+
+		// Add field with default value as expression builder
+
+		String expressionBuilderDefaultValue = String.format(
+			"\"%s\"", _listTypeEntryKey);
+
+		objectField = _addCustomObjectField(
+			new PicklistObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).listTypeDefinitionId(
+				_listTypeDefinition.getListTypeDefinitionId()
+			).name(
+				"a" + RandomTestUtil.randomString()
+			).objectDefinitionId(
+				objectDefinition.getObjectDefinitionId()
+			).objectFieldSettings(
+				Arrays.asList(
+					_createObjectFieldSetting(
+						ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+						expressionBuilderDefaultValue),
+					_createObjectFieldSetting(
+						ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+						ObjectFieldSettingConstants.VALUE_EXPRESSION_BUILDER))
+			).build());
+
+		_assertObjectFieldSettingDefaultValue(
+			expressionBuilderDefaultValue,
+			ObjectFieldSettingConstants.VALUE_EXPRESSION_BUILDER, objectField);
+
+		_assertDefaultValue(_listTypeEntryKey, objectField);
+
+		// Update field's default value
+
+		objectField.setObjectFieldSettings(
+			Arrays.asList(
+				_createObjectFieldSetting(
+					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+					_listTypeEntryKey),
+				_createObjectFieldSetting(
+					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+					ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE)));
+
+		objectField = _updateCustomObjectField(objectField);
+
+		_assertObjectFieldSettingDefaultValue(
+			_listTypeEntryKey, ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE,
+			objectField);
+
+		_assertDefaultValue(_listTypeEntryKey, objectField);
+
+		objectField.setObjectFieldSettings(
+			Arrays.asList(
+				_createObjectFieldSetting(
+					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+					expressionBuilderDefaultValue),
+				_createObjectFieldSetting(
+					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+					ObjectFieldSettingConstants.VALUE_EXPRESSION_BUILDER)));
+
+		objectField = _updateCustomObjectField(objectField);
```

Comment:
@devpedrotavares you can add `objectField.setObjectFieldSettings()` as parameter in `_updateCustomObjectField`.

Reply:
Hey @murilostodolni, don't you think we should have the `_updateCustomObjectField` be generic?

Reply:
Yes, the `_updateCustomObjectField` should be generic. You can do something like this:

```
picklistObjectField = _updateCustomObjectField(picklistObjectField.setObjectFieldSettings(
			Arrays.asList(
				_createObjectFieldSetting(
					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
					_listTypeEntryKey),
				_createObjectFieldSetting(
					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
					ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE))));
```

Reply:
You don't need change the method, just inline.